### PR TITLE
docs: adds intent of Validation and Mutation

### PR DESCRIPTION
**Pull request by nilekhc**
Signed-off-by: Nilekh Chaudhari <1626598+nilekhc@users.noreply.github.com>

**What this PR does / why we need it**:

Ref - https://github.com/open-policy-agent/gatekeeper-library/pull/285#discussion_r1089590219

**Which issue(s) does this PR fix** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by sozercan — APPROVED** [review]

**Comment by nilekhc** [comment]
@maxsmythe Plz let me know if there is anything I need to update here. Thanks!

File: README.md [hunk 1]
```diff
@@ -3,6 +3,11 @@
 
 A community-owned library of policies for the [OPA Gatekeeper project](https://open-policy-agent.github.io/gatekeeper/website/docs/).
 
+## Validation and Mutation
+The library consists of two main components: `Validation` and `Mutation`.
+- Validation: Gatekeeper validates the resources in the cluster against the policies defined in the library. The policies are defined as `ConstraintTemplates` and `Constraints`. The `ConstraintTemplates` define the policy logic and the `Constraints` define the parameters for the policy.
```
Comment (maxsmythe) [inline README.md:8]: Something like: "constraint templates" can be applied directly to a cluster. You then apply "constraints" to customize policy to fit your specific needs.

For mutation: "Mutation policies are only examples, they should be customized to meet your needs before being applied"

**Comment by maxsmythe** [comment]
Sorry for the slow response

**Comment by nilekhc** [comment]
> Sorry for the slow response

@maxsmythe, no worries. I have updated the wording. PTAL when you get a chance.

File: README.md [hunk 2]
```diff
@@ -3,6 +3,11 @@
 
 A community-owned library of policies for the [OPA Gatekeeper project](https://open-policy-agent.github.io/gatekeeper/website/docs/).
 
+## Validation and Mutation
+The library consists of two main components: `Validation` and `Mutation`.
+- Validation: Gatekeeper validates the resources in the cluster against the policies defined in the library. The policies are defined as `ConstraintTemplates` and `Constraints`. `ConstraintTemplates` can be applied directly to a cluster and then `Constraints` can be applied to customize policy to fit your specific needs.
```
Comment (ritazh) [inline README.md:8]:
```suggestion
- Validation: Gatekeeper can validate resources in the cluster against Gatekeeper policies, such as the policies defined in the library. The policies are defined as `ConstraintTemplates` and `Constraints`. `ConstraintTemplates` can be applied directly to a cluster and then `Constraints` can be applied to customize policy to fit your specific needs.
```

File: README.md [hunk 3]
```diff
@@ -3,6 +3,11 @@
...
+- Mutation: Gatekeeper mutates the resources in the cluster against the policies defined in the library. Mutation policies are only examples, they should be customized to meet your needs before being applied.
```
Comment (ritazh) [inline README.md:9]:
```suggestion
- Mutation: Gatekeeper can mutate resources in the cluster against the Gatekeeper mutation policies, such as these defined in the library. Mutation policies are only examples, they should be customized to meet your needs before being applied.
```

**Comment by nilekhc** [comment]
@maxsmythe @ritazh, please let me know if there needs to be any modification. Thanks.

**Review by maxsmythe — APPROVED** [review]
LGTM

**Review by ritazh — APPROVED** [review]
LGTM

**Pull request merged**